homeassistant/components/sensor/yr.py

```diff
@@ -0,0 +1,234 @@
+"""
+homeassistant.components.sensor.yr
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Yr.no weather service.
+
+Configuration:
+
+Will show a symbol for the current weather as default:
+sensor:
+  platform: yr
+
+Will show temperatue and wind direction:
+sensor:
+  platform: yr
+  monitored_conditions:
+    - temperature
+    - windDirection
+
+Will show all available sensors:
+sensor:
+  platform: yr
+  monitored_conditions:
+    - temperature
+    - symbol
+    - precipitation
+    - windSpeed
+    - pressure
+    - windDirection
+    - humidity
+    - fog
+    - cloudiness
+    - lowClouds
+    - mediumClouds
+    - highClouds
+    - dewpointTemperature
+
+"""
+import logging
+import datetime
+import urllib.request
+import xmltodict
```

Comment:
Don't import requirement dependencies on the top of the file but move it inside the methods that use it.